torch/_dynamo/metrics_context.py

```diff
@@ -0,0 +1,124 @@
+import time
+from contextlib import contextmanager
+from typing import Any, Callable, Dict, Generator, Optional, Type, Union
+from typing_extensions import TypeAlias
+
+
+OnExitType: TypeAlias = Callable[
+    [Dict[str, Any], Optional[Type[BaseException]], Optional[BaseException]], None
+]
+
+
+class MetricsContext:
```

Comment:
Re stacking or not, there's no reason to keep this stacked separately. I /want/ this landed with its use sites because otherwise I have no way of evaluating if it's working in practice.